velox/expression/tests/CastExprTest.cpp

```diff
@@ -1700,6 +1700,30 @@ TEST_F(CastExprTest, integerToDecimal) {
   testIntToDecimalCasts<int64_t>();
 }
 
+TEST_F(CastExprTest, boolToDecimal) {
+  // Bool to short decimal.
+  auto input =
+      makeFlatVector<bool>({true, false, false, true, true, true, false});
+  testComplexCast(
```

Comment:
Curious why is decimal a complex type?

Reply:
Seems `testComplexCast` is not restricted to complex type, but covers more cases than `testCast`, including constant input and dictionary input.